cql3/query_processor.cc

```diff
@@ -74,6 +76,8 @@ query_processor::query_processor(service::storage_proxy& proxy, service::forward
         , _authorized_prepared_cache_update_interval_in_ms_observer(_db.get_config().permissions_update_interval_in_ms.observe(_auth_prepared_cache_cfg_cb))
         , _authorized_prepared_cache_validity_in_ms_observer(_db.get_config().permissions_validity_in_ms.observe(_auth_prepared_cache_cfg_cb))
         , _wasm_engine(std::move(wasm_engine))
+        , _wasm_instance_cache(instance_cache.local_is_initialized() ? &instance_cache.local() : nullptr)
```

Comment:
So query_processor wants a reference on instance_cache instance (not sharded<instance_cache> thing). And no other services need instance_cache instance. Can the instance_cache be removed from main (and cql_test_env) in favor of query_processor initializing it on its own? Via e.g. unique_ptr or optional

Reply:
Yes, though we'll have to pass all the arguments for the cache constructor to the query_processor. Maybe I can wrap them together with engine in a single struct - I'll try that in a rebase.